tests/0148-offset_fetch_commit_error_mock.c

```diff
@@ -0,0 +1,574 @@
+                                 "bootstrap.servers", bootstraps, NULL);
+
+        /* Consume same messages */
+        consumer = test_create_consumer(topic, NULL, conf, NULL);
+        test_consumer_subscribe(consumer, topic);
+        test_msgver_init(&mv, testid);
+        test_consumer_poll("receive first batch", consumer, testid, -1, 0,
+                           msgcnt, &mv);
+        test_msgver_verify("verify first batch", &mv, TEST_MSGVER_ALL, 0,
+                           msgcnt);
+        test_msgver_clear(&mv);
+
+        /* Set OffsetCommit errors */
+        rd_kafka_mock_push_request_errors(mcluster, RD_KAFKAP_OffsetCommit, 1,
+                                          expected_err);
+
+        if (variation ==
+            TEST_MANUAL_COMMIT_ERROR_VARIATION_STORE_OFFSET_MANUALLY) {
+                /* Variation 1: pass offsets to commit */
+                to_commit = rd_kafka_topic_partition_list_new(1);
+                rd_kafka_topic_partition_list_add(to_commit, topic, 0)->offset =
+                    msgcnt;
+        }
+
+        /* Sync commit */
+        err = rd_kafka_commit(consumer, to_commit, rd_false);
+        TEST_ASSERT(err == expected_err, "Expected error %s, got %s",
+                    rd_kafka_err2name(expected_err), rd_kafka_err2name(err));
+
+        /* Retry it, this time it should work */
+        err = rd_kafka_commit(consumer, to_commit, rd_false);
+        TEST_ASSERT(err == RD_KAFKA_RESP_ERR_NO_ERROR,
+                    "Expected error %s, got %s",
+                    rd_kafka_err2name(RD_KAFKA_RESP_ERR_NO_ERROR),
+                    rd_kafka_err2name(err));
+
+        RD_IF_FREE(to_commit, rd_kafka_topic_partition_list_destroy);
+
+        rd_kafka_destroy(consumer);
+
+        SUB_TEST_PASS();
+}
+
+/**
+ * Scenarios:
+ * during_revocation: the auto-commit is triggered by a revocation,
+ *                    otherwise it's triggered by the consumer close.
+ * session_times_out: Session times out giving UNKNOWN_MEMBER_ID, otherwise
+ *                    commit succeeds after last STALE_MEMBER_EPOCH. When
+ *                    session times out the auto-commit fails and messages
+ *                    are consumed again.
+
+ * Variations:
+ *    - TEST_AUTO_COMMIT_STALE_MEMBER_EPOCH_VARIATION_NO_REVOKE_NO_TIMEOUT:
+ *          during_revocation=false, session_times_out=false
+ *    - TEST_AUTO_COMMIT_STALE_MEMBER_EPOCH_VARIATION_NO_REVOKE_WITH_TIMEOUT:
+ *          during_revocation=false, session_times_out=true
+ *    - TEST_AUTO_COMMIT_STALE_MEMBER_EPOCH_VARIATION_REVOKE_NO_TIMEOUT:
+ *          during_revocation=true, session_times_out=false
+ *    - TEST_AUTO_COMMIT_STALE_MEMBER_EPOCH_VARIATION_REVOKE_WITH_TIMEOUT:
+ *          during_revocation=true, session_times_out=true
+ */
+typedef enum do_test_OffsetCommit_automatic_stale_member_epoch_error_variation_t {
+        TEST_AUTO_COMMIT_STALE_MEMBER_EPOCH_VARIATION_NO_REVOKE_NO_TIMEOUT = 0,
```

Comment:
I was suggesting to add documentation to each variation and removing it from previous list for example:
```suggestion
        /** during_revocation=false, session_times_out=false */
        TEST_AUTO_COMMIT_STALE_MEMBER_EPOCH_VARIATION_NO_REVOKE_NO_TIMEOUT = 0,
```

same for the other variation